python/lsst/pipe/supertask/activator.py

```diff
@@ -131,6 +132,34 @@ def activate(self):
         """
         pass
 
+def get_inheritance(stask_class, subclasses=[]):
+    """
+    Get all inheritance classes for a given task to differentiate
+     between SuperTask, CmdLineTask, etc.
+
+    :param stask_class: the output class from pyclbr containing the sub-class information
+    :return: a string with all sub-classes
+    """
+
+    temp_class = stask_class.super[0] # simple inheritance
+    if isinstance(temp_class, pyclbr.Class):
+        subclasses.append(temp_class.name)
+        subclasses = get_inheritance(temp_class, subclasses)
+    else:
+        subclasses.append(temp_class)
+    return subclasses
+
+
+class mock_class(object):
```

Comment:
I don't like this name. `mock_class` suggests to me something that can be inserted as a placeholder to provide stubbed implementations. This class really seems to be somewhere to store an externally supplied Python search path for locating super tasks. Something like `class CmdLineOverride(object):` ?  And rather than `self.mock` use `self.override`. Or else `class ManualPath():`?